commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareReportServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package org.motechproject.commcare.service.impl;
+
+import com.google.common.reflect.TypeToken;
+import org.motechproject.commcare.client.CommCareAPIHttpClient;
+import org.motechproject.commcare.domain.CommcareMetadataInfo;
+import org.motechproject.commcare.domain.CommcareMetadataJson;
+import org.motechproject.commcare.domain.report.ReportMetadataInfo;
+import org.motechproject.commcare.domain.report.ReportsMetadataInfo;
+import org.motechproject.commcare.domain.report.ReportMetadataJson;
+import org.motechproject.commcare.domain.report.ReportsMetadataResponseJson;
+import org.motechproject.commcare.domain.report.constants.ColumnType;
+import org.motechproject.commcare.domain.report.constants.FilterDataType;
+import org.motechproject.commcare.domain.report.constants.FilterType;
+import org.motechproject.commcare.service.CommcareConfigService;
+import org.motechproject.commcare.service.CommcareReportService;
+import org.motechproject.commons.api.json.MotechJsonReader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * A {@link CommcareReportService} that is responsible for interacting with CommCareHQ's Report Metadata and Data API.
+ */
+@Service
+public class CommcareReportServiceImpl implements CommcareReportService {
+
+    private CommCareAPIHttpClient commCareHttpClient;
+    private CommcareConfigService configService;
+    private MotechJsonReader motechJsonReader;
+
+    @Autowired
+    public CommcareReportServiceImpl(CommCareAPIHttpClient commCareHttpClient, CommcareConfigService configService) {
+        this.commCareHttpClient = commCareHttpClient;
+        this.configService = configService;
+        this.motechJsonReader = new MotechJsonReader();
+    }
+
+    @Override
+    public ReportsMetadataInfo getReportsList(String configName) {
+        String response = commCareHttpClient.reportsListMetadataRequest(configService.getByName(configName).getAccountConfig());
+
+        ReportsMetadataResponseJson reportsMetadataResponseJson = parseReportsFromResponse(response);
+
+        return new ReportsMetadataInfo(generateReportsFromReportsResponse(reportsMetadataResponseJson.getReports()), populateReportsMetadata(reportsMetadataResponseJson.getMetadata()));
+    }
+
+    @Override
+    public ReportsMetadataInfo getReportsList() {
+        return getReportsList(null);
```

Comment:
null ? it looks strange. If configName is null in getByName method then the default value is used ?